### PR TITLE
Add defensibility and tech investment

### PR DESCRIPTION
**Pull request by malomarrec**
I think we are missing a section explaining how we invest in immediate value features, but also have a long-term strategy to build foundational technology to explore and manipulate the Sourcegraph.
In practice: the day Code Intel and Compute are available as a platform to, say, create batch changes, we will be able to unlock a wow! use case AND this will be hard to copy because it is a lot of foundational work and a lot of longing operational work to provide this layer.

Examples of long-term applications include:
- easily defining arbitrarily complex Code Insights
- accurately rename a symbol with a batch change across dependent repositories

### CODE REVIEW

File: content/strategy-goals/strategy/code-graph/index.md [hunk 1]
```diff
@@ -49,3 +49,7 @@ We see Sourcegraph playing an important role in this emerging need for complexit
 If we achieve these, we make it easy for developers to get things done and, ultimately, ship great code.
 
 [1]: https://future.a16z.com/the-case-for-developer-experience/
+
+## Technology and defensibility
+
+On top of building features that are immediately useful to customers, Code graph is investing in key technology areas that will drive Sourcegraph's long term product differentiation and defensibility. Code Intelligence's technology and indexing capabilities, as well as Code Search's compute platform will provide the foundations for delivering a unique, hard to replicate experience across use cases and features, allowing developers to navigate, understand and manipulate their Code Graph.
```
Comment (jasonyavorsky) [inline content/strategy-goals/strategy/code-graph/index.md]: Could this be stated simply in one of the principles in the section above? Maybe:

- 🔬 **Innovative**: Investing in underlying search technologies to manipulate the code graph helps us differentiate as well as defend our leading market position.

I like the addition, I'd just like to see it integrated into our existing content. A new section at the bottom could be easy to miss.

**Review by jasonyavorsky — APPROVED** [review]

File: content/strategy-goals/strategy/code-graph/index.md (hunk 1 above)
Comment (malomarrec) [reply content/strategy-goals/strategy/code-graph/index.md]: Makes sense, thank you! I just committed a proposal.

Comment (scalabilitysolved) [reply content/strategy-goals/strategy/code-graph/index.md]: Do we consider here Code Intelligence to fit in the "underlying search technologies"?

Comment (malomarrec) [reply content/strategy-goals/strategy/code-graph/index.md]: I'm not sure I understand your question, but my take is that "Code Intelligence the team" is building two things:
- an immediately useful feature called Code Intelligence
- the underlying technology (including indexers) that, as a platform, can be used by multiple product areas and is hard for someone else to replicate.
Is that a mischaracterization?

**Comment by scalabilitysolved** [comment]
That's correct on both fronts.  My question was more whether we were
classifying it as falling under the "search technologies" umbrella from the
comment by Jason.

> Hard for someone else to replicate

It's hard but not impossible (especially as how we've solved the problems
are public).  There are definitely some fundamental strategy differences
though compared to competitors, we're favouring accuracy over availability.

**Review by lguychard — APPROVED** [review]

**Comment by jasonyavorsky** [comment]
@malomarrec please share in the code graph channel after merging

**Pull request merged**